makefiles/kconfig.mk

```diff
@@ -26,6 +26,9 @@ export KCONFIG_AUTOHEADER_HEADER
 # This file will contain the calculated dependencies formated in Kconfig
 export KCONFIG_GENERATED_DEPENDENCIES = $(GENERATED_DIR)/Kconfig.dep
 
+# This file will contain external module configurations
+export KCONFIG_EXTERNAL_CONFIGS = $(GENERATED_DIR)/Kconfig.external_modules
```

Comment:
These could be conditionally export, same for the above, but I would rather do both as a cleanup in a followup.